### PR TITLE
Serving OR Queries from the client-side index.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java [hunk 1]
```diff
@@ -325,6 +327,16 @@ public IndexType getIndexType(Target target) {
         result = IndexType.PARTIAL;
       }
     }
+
+    // OR queries have more than one sub-target (one sub-target per DNF term). We currently consider
+    // all OR queries to have partial indexes, and hence do sorting and limit in post-processing.
+    // TODO(orquery): If we have a FULL index *and* we have the index that can be used for sorting
+    //  all DNF branches on the same value, we can improve performance by performing a JOIN in SQL.
+    //  See b/235224019 for more information.
+    if (subTargets.size() > 1 && result == IndexType.FULL) {
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java]: Is it possible to add a test case for the new change then?

File: firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java [hunk 2]
```diff
@@ -1007,90 +1008,148 @@ public void testPartialIndexAndFullIndex() throws Exception {
...
+            .orderBy(orderBy("a"));
+    validateIndexType(query7, IndexManager.IndexType.FULL);
+
+    // OR query with implicit orderBy without limit which has all sub-target indexes.
+    Query query8 = query("coll").filter(orFilters(filter("a", ">", 1), filter("b", "==", 1)));
+    validateIndexType(query8, IndexManager.IndexType.FULL);
+
+    // OR query without orderBy with limit which has all sub-target indexes.
+    Query query9 =
+        query("coll").filter(orFilters(filter("a", "==", 1), filter("b", "==", 1))).limitToFirst(2);
```
Comment (wu-hui) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java:1133]: Would recommend mixing limitToLast in some of the cases here. Optional.

**Review by wu-hui — APPROVED** [review]
LGTM! Thanks!!

File: firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java (hunk 2 above)
Comment (ehsannas) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java:1134]: Done

**Pull request merged**